src/components/CartModal/CartModal.tsx

```diff
@@ -0,0 +1,56 @@
+import { useCartManagement } from '../../hooks/useCartManager';
+import useGetCarts from '../../hooks/useGetCarts';
+import { useModalClose } from '../../hooks/useModalClose';
+import CartItem from './CartItem';
+import {
+  CloseButton,
+  EmptyCart,
+  ModalContainer,
+  ModalOverlay,
+  ModalTitle,
+} from './CartModal.style';
+import TotalPrice from './TotalPrice';
+
+interface ModalProps {
+  onClose: () => void;
+}
+
+function CartModal({ onClose }: ModalProps) {
+  const { carts, cartItemCount, refetchCarts } = useGetCarts();
+  const { modifyCartItem, deleteItemFromCart } = useCartManagement({
+    cartItemCount,
+    carts,
+    refetchCarts,
+  });
```

Comment:
useGetCarts 와 useCartManagement 를 CartModal 이 아닌 CartItem 내부에서 사용하는 식으로 구성하여 CartModal 내부의 동작과 책임을 단순화 시킬 수 있을 것 같은데요, 메리는 어떻게 생각하시나요?

```suggestion
  const { carts } = useGetCarts();
```

Reply:
크리스가 제안해주신 방법에 대해서 생각해봤는데, 현재 구조를 유지하는 것이 적합할 것 같다는 생각이 들었습니다.
제가 `CartItem`이 아닌 `CartModal` 컴포넌트의 내부에서 2개의 커스텀 훅으로 데이터를 받아온 이유는 아래와 같습니다!

1. 현재 `price`라는 변수로 **전체 가격**을 계산해주고 있는데, 이때 `useGetCarts`를 통해 받아온 `carts` 데이터가 필요했습니다.
2. 빈 장바구니 `UI`를 처리하기 위해서도 전체 `carts`의 길이가 필요했습니다.

크리스가 제안해주신 방법에 대해서 생각해봤는데, 만약 `useGetCarts`를 `CarItem` 내부에서 사용하게 된다면 `CartItem` 컴포넌트를 생성할 때마다 `api` 요청이 생길 텐데 그럼 `CartItem`이 늘어날 수록 성능적인 이슈가 생기지는 않을까? 라는 생각이 들었습니다. 

크리스가 언급해주신 문제로 `CartModal`의 책임이 복잡해진다는 문제가 있긴 하지만, 현재 내부 로직을 2개의 커스텀 훅으로 분리했기 때문에 그렇게까지 많은 로직이 얽혀있다고 생각이 되지 않았습니다.
혹시 제가 놓친 부분이 있다면 추가 코멘트 부탁드립니다!

Reply:
useGetCarts 를 CartItem 내부에서 사용하더라도 key 를 기반으로 동일한 상태값을 그대로 사용할 수 있는 지금 구조상 API 요청이 CartItem 컴포넌트의 개수만큼 생겨나지는 않을 것 같은데요.

다만 말씀하신 것처럼 useGetCarts 를 CartItem 컴포넌트 내부에서 사용하는 것이 그렇지 않는 방법보다 리렌더링을 더 많이 트리거 할 수 있기에, 성능적 측면에서는 지금의 방식이 나을 수 있다는 생각은 드네요.(다만 저는 그것이 서비스 품질에 영향을 줄 정도의 성능 하락으로 이어지리라고는 생각하지 않기는 합니다.) 말씀하신 것처럼 지금 구조도 크게 복잡한 것은 아니라, 지금 구조를 유지하셔도 좋다고 생각합니다!